### PR TITLE
Support Appium mobile selectors

### PR DESCRIPTION
**Pull request by amitzur**
[Trello](https://trello.com/c/3ZNabDwo)

This is a first attempt to implement region checkpoints for native apps.
The gist of it is to call `el.getSize()` and `el.getLocation` on the target element when it's native mobile, instead of trying to execute JS.

This includes a small refactor to `_updateScalingParams` intended only for better readibility.
It also includes caching of `driver.controller.isNative()` since it's run so many times.

Still open issues:
1. [x] I had to remove `extractElement` from `SpecWrappedElement` since otherwise I wouldn't have the `getSize` and `getLocation` methods. But this creates an issue for other cases. I couldn't remember what's the purpose of it.
2. [x] Support wdio4.
3. [x] The coverage test is not fully automatic. Appium server needs to be up with AVD and predefined device. Maybe we can leverage Sauce here. I wasn't sure if we can upload the APK there as part of capabilities.
4. [ ] Run the automated test only in wdio4 and wdio5

### CODE REVIEW

File: packages/eyes-sdk-core/lib/EyesCore.js [hunk 1]
```diff
@@ -464,47 +464,63 @@ class EyesCore extends EyesBase {
...
-    this._logger.verbose('Done!')
-    return factory
+        devicePixelRatio = EyesUtils.getMobilePixelRatio(this._logger, this._driver, viewportSize)
+      } else {
+        devicePixelRatio = EyesUtils.getDevicePixelRatio(this._logger, this._driver)
+      }
+    } catch (err) {
+      this._logger.verbose('Failed to extract device pixel ratio! Using default.', err)
+      devicePixelRatio = DEFAULT_DEVICE_PIXEL_RATIO
+    }
```
Comment (kyryloonufriiev) [inline packages/eyes-sdk-core/lib/EyesCore.js:486]: This is really the much better way to do it. And also prevent us from call API that not exists.

Comment (amitzur) [reply packages/eyes-sdk-core/lib/EyesCore.js:486]: glad to hear :)

File: packages/eyes-webdriverio-5/src/SpecWrappedElement.js [hunk 2]
```diff
@@ -70,7 +76,9 @@ module.exports = {
   toSupportedSelector,
   toEyesSelector,
   extractId,
-  extractElement,
+  // extractElement,
```
[the remark this answers is not in the document]
Comment (amitzur) [reply packages/eyes-webdriverio-5/src/SpecWrappedElement.js:79]: So how do you propose we solve this tradeoff. On one side, we need to peel off the extended element for IE and Edge. On the other hand, we need it to access `getSize` and `getLocation` for native.

File: packages/eyes-sdk-core/lib/EyesUtils.js [hunk 3]
```diff
@@ -333,8 +332,15 @@ async function getElementClientRect(_logger, executor, element) {
  * @param {EyesWrappedElement} element - element to get rect
  * @return {Promise<Region>} element rect
  */
-async function getElementRect(_logger, executor, element) {
-  const rect = await executor.executeScript(EyesJsSnippets.GET_ELEMENT_RECT, element)
+async function getElementRect({logger: _logger, executor, element, isNative}) {
+  let rect
+  if (isNative) {
+    const {x, y} = await element.spec.getNativeLocation(element.unwrapped)
```
[the remark this answers is not in the document]
Comment (amitzur) [reply packages/eyes-sdk-core/lib/EyesUtils.js:338]: Let's discuss on Monday. We need to finalize this next week and I'd like to hear your thoughts also in light of the work on playwright, if it has any influence on it.

File: packages/eyes-webdriverio-5/src/SpecWrappedElement.js [hunk 4]
```diff
@@ -62,6 +62,12 @@ function isStaleElementReferenceResult(result) {
   const errOrResult = result.originalError || result
   return errOrResult instanceof Error && errOrResult.name === 'stale element reference'
 }
+function getNativeLocation(element) {
+  return element.getLocation()
+}
+function getNativeSize(element) {
+  return element.getSize()
+}
```
[the remark this answers is not in the document]
Comment (amitzur) [reply packages/eyes-webdriverio-5/src/SpecWrappedElement.js:70]: ah, that answers my question below. You're saying we can detect if element is extended and if not call something like `extendElement` if it exists on the spec (i.e. `driver.$`).
Did I get you correctly?

**Comment by amitzur** [comment]
I tried to use sauce labs, but they don't seem to load the app correctly. Or I'm doing something wrong.

**Comment by amitzur** [comment]
@kyryloonufriiev we only need to discuss filtering of coverage tests

**Pull request merged**